clang/test/dpct/helper_files_ref/include/ccl_utils.hpp

```diff
@@ -32,32 +35,226 @@ get_kvs(const oneapi::ccl::kvs::address_type &addr) {
+        _comm(oneapi::ccl::create_communicator(
+            size, rank, _device_comm, _context_comm, dpct::ccl::create_kvs(id),
+            attr)) {}
+
+  ~communicator_ext(){};
+
+  /// Return the rank in a oneapi::ccl::communicator
+  /// \returns The rank corresponding to communicator object
+  int rank() const {
+    return _comm.rank();
+  }
+
+  /// Retrieves the number of rank in oneapi::ccl::communicator
+  /// \returns The number of the ranks
+  int size() const {
+    return _comm.size();
+  }
+
+  /// Return underlying native device, which was used in oneapi::ccl::communicator
+  sycl::device get_device() const {
+    return _comm.get_device().get_native();
+  }
+
+  /// \brief allreduce is a collective communication operation that performs the global reduction operation
+  ///       on values from all ranks of communicator and distributes the result back to all ranks.
+  /// \param send_buf the buffer with @c count elements of @c dtype that stores local data to be reduced
+  /// \param recv_buf [out] the buffer to store reduced result, must have the same dimension as @c send_buf
+  /// \param count the number of elements of type @c dtype in @c send_buf and @c recv_buf
+  /// \param dtype the datatype of elements in @c send_buf and @c recv_buf
+  /// \param rtype the type of the reduction operation to be applied
+  /// \param stream a sycl::queue ptr associated with the operation
+  /// \return @ref void
+  void allreduce(const void *sendbuff, void *recvbuff, size_t count,
+                 dpct::library_data_t dtype, oneapi::ccl::reduction rtype,
+                 sycl::queue *stream) const {
+    ccl_func_adapter(
+        [=](const oneapi::ccl::stream &queue) {
+          return oneapi::ccl::allreduce(
+              sendbuff, recvbuff, count,
+              dpct::ccl::detail::to_ccl_datatype(dtype), rtype, _comm, queue);
+        },
+        stream);
+  }
+
+  /// \brief reduce is a collective communication operation that performs the
+  ///        global reduction operation on values from all ranks of the communicator
+  ///        and returns the result to the root rank.
+  /// \param send_buf the buffer with @c count elements of @c dtype that stores
+  ///        local data to be reduced 
+  /// \param recv_buf [out] the buffer to store reduced result, 
+  ///        must have the same dimension as @c send_buf 
+  /// \param count the number of elements of type @c dtype in @c send_buf and @c recv_buf 
+  /// \param dtype the datatype of elements in @c send_buf and @c recv_buf 
+  /// \param root the rank that gets the result of reduction 
+  /// \param rtype the type of the reduction operation to be applied 
+  /// \param stream a sycl::queue ptr associated with the operation 
+  /// \return @ref void
+  void reduce(const void *sendbuff, void *recvbuff, size_t count,
+              dpct::library_data_t dtype, oneapi::ccl::reduction rtype,
+              int root, sycl::queue *stream) const {
```

Comment:
Can we rename this variable `stream` -> `queue`

Reply:
Can we use a mutable reference here? but reference may make the migration stage more complex, what do you all think about?